PFERD/crawl/ilias/ilias_web_crawler.py

```diff
@@ -35,12 +45,32 @@ def base_url(self) -> str:
 
         return base_url
 
-    def client_id(self) -> str:
-        client_id = self.s.get("client_id")
-        if not client_id:
-            self.missing_value("client_id")
-
-        return client_id
+    def login(self) -> Union[ShibbolethLoginType, LocalLoginType]:
```

Comment:
```suggestion
    def login(self) -> Union[Literal["shibboleth"], LocalLoginType]:
```

I am not sure the dummy type is nicer than a literal?